syntax = "proto3";

import "google/protobuf/any.proto";
import "enums.proto";

package pslx;

// the next will be 4
message NodeSnapshot {
    string node_name = 1;
    repeated string children_names = 2;
    repeated string parents_names = 3;
}

// the next will be 10
message OperatorSnapshot {
    string operator_name = 1;
    DataModelType data_model = 2;
    Status status = 3;
    NodeSnapshot node_snapshot = 4;
    string start_time = 5;
    string end_time = 6;
    google.protobuf.Any content = 7;
    string class_name = 8;
    string log_file = 9;
}

// the next will be 13
message ContainerSnapshot {
    string container_name = 1;
    map<string, OperatorSnapshot> operator_snapshot_map = 2;
    bool is_initialized = 3;
    Status status = 4;
    string start_time = 5;
    string end_time = 6;
    string class_name = 7;
    ModeType mode = 8;
    DataModelType data_model = 9;
    string log_file = 10;
    map<string, int32> counters = 11;
    string run_cell = 12;
}

// the next will be 6
message OperatorContentPlain {
    oneof plan_content {
        string string_val = 1;
        int32 int_val = 2;
        float float_val = 3;
        bool bool_val = 4;
        bytes bytes_val = 5;
    }
}
